### DISCUSSION TITLE
How to get the inner editor of a math node when i call  a command?

### DISCUSSION THREAD

**Discussion by 4-1-1** [Milkdown/milkdown#798] [Q&A]
I made a math template. so I will insert data to mathNode value. 
But I can not insert template content into the inner editor. Just append content to the value. 
Is there a way to insert content to the cursor
```typescript
  createCmd(InsertMath, (attrs) => (state, dispatch) => {
                const text = attrs?.value || '';
                const isNodeSelection = state.selection instanceof NodeSelection;
                const node = isNodeSelection ? (state.selection as NodeSelection).node : null;
                let currentValue = '';
                if (node && node.type === nodeType) {
                    const view = ctx.get(editorViewCtx);
                    const dom = view.domAtPos(state.selection.from + 1);
                    if (dom.node) {
                        const dataDiv = (dom.node as HTMLDivElement).querySelector('[data-type="math_block"]');
                        if (dataDiv) {
                            currentValue = (dataDiv as HTMLDivElement).dataset['value'] || '';
                        }
                    }
                }
                const nextValue = currentValue + text;
                const tr = state.tr;
                const nextTr = tr.replaceSelectionWith(
                    nodeType.create({ value: nextValue }, state.schema.text(nextValue || ' ')),
                );
                if (node) {
                    const rs = nextTr.doc.resolve(nextTr.mapping.map(state.selection.from));

                    nextTr.setSelection(new NodeSelection(rs)).scrollIntoView();
                }
                const pos: number = state.selection.to;
                dispatch?.(nextTr);
                if (!node) {
                    const rs = nextTr.doc.resolve(nextTr.mapping.map(pos));
                    setTimeout(() => {
                        dispatch?.(nextTr.setSelection(new NodeSelection(rs)));
                    }, 100);
                }
                return true;
            }),
```

**Comment by Saul-Mirone** [comment]
If that's the case, why don't create math node with default value?

**Reply by 4-1-1** [reply]
@Saul-Mirone Did you mean if i have select a math node, when i use a math template, Insert a new math node after the selected?
I think it's not fit the user's expect

**Reply by Saul-Mirone** [reply]
You can get the value of the math node through `node.attr.value`. And you can apply you changes by calling [`tr.setNodeMarkup`](https://prosemirror.net/docs/ref/#transform.Transform.setNodeMarkup).

**Reply by 4-1-1** [reply]
> You can get the value of the math node through `node.attr.value`. And you can apply you changes by calling [`tr.setNodeMarkup`](https://prosemirror.net/docs/ref/#transform.Transform.setNodeMarkup).

In this way , I can just get the old value, and insert new value to the end.  not to the position of cursor.

May I should use a react component to do this

**Reply by Saul-Mirone** [reply]
I don't think we have a way to get the inner editor for now. Maybe we need to add an ctx for this. Something like `MathBlockInnerEditorCtx`. And we also need to return [inner editor](https://github.com/Saul-Mirone/milkdown/blob/main/packages/theme-pack-helper/src/renderer-preset/inner-editor.ts#L148) in the renderer to make it can be stored somewhere.

A react component maybe also can solve your problem. You'll take over the control of the node view for math block.

**Reply by 4-1-1** [reply]
> I don't think we have a way to get the inner editor for now. Maybe we need to add an ctx for this. Something like `MathBlockInnerEditorCtx`. And we also need to return [inner editor](https://github.com/Saul-Mirone/milkdown/blob/main/packages/theme-pack-helper/src/renderer-preset/inner-editor.ts#L148) in the renderer to make it can be stored somewhere.
> 
> A react component maybe also can solve your problem. You'll take over the control of the node view for math block.

I did it with  "document.execCommand". and It works fine in chrome.  hooray!
